### PR TITLE
KAFKa-16331: remove EOSv1 support for getting clientInstanceIds

### PR DESCRIPTION
**Pull request by mjsax**
With EOSv1 removed, we don't need to collect producer's clientInstanceIds per task any longer. While we never completed this feature, we can remove the corresponding scaffolding code.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java [hunk 1]
```diff
@@ -1877,12 +1877,10 @@ public synchronized ClientInstanceIds clientInstanceIds(final Duration timeout)
         // (1) fan-out calls to threads
 
         // StreamThread for main/restore consumers and producer(s)
-        final Map<String, KafkaFuture<Uuid>> consumerFutures = new HashMap<>();
-        final Map<String, KafkaFuture<Map<String, KafkaFuture<Uuid>>>> producerFutures = new HashMap<>();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java:1881]: Originally, we did split consumer and producer to be able to handle the producer-per-task case. Now, we only have producer-per-thread and thus can unify consumer/producer into "clients"

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java [hunk 2]
```diff
@@ -1907,65 +1905,30 @@ public synchronized ClientInstanceIds clientInstanceIds(final Duration timeout)
...
+                    clientFuture.getKey(),
                     instanceId
                 );
             } else {
-                log.debug(String.format("Telemetry is disabled for %s.", consumerFuture.getKey()));
+                log.debug(String.format("Telemetry is disabled for %s.", clientFuture.getKey()));
             }
         }
 
-        // (3b) collect producers from StreamsThread
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java:1933]: This case is now covered with 3(a) step

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 3]
```diff
@@ -349,8 +349,7 @@ public boolean isStartingRunningOrPartitionAssigned() {
     private volatile long fetchDeadlineClientInstanceId = -1;
     private volatile KafkaFutureImpl<Uuid> mainConsumerInstanceIdFuture = new KafkaFutureImpl<>();
     private volatile KafkaFutureImpl<Uuid> restoreConsumerInstanceIdFuture = new KafkaFutureImpl<>();
-    private volatile KafkaFutureImpl<Map<String, KafkaFuture<Uuid>>> producerInstanceIdFuture = new KafkaFutureImpl<>();
-    private volatile KafkaFutureImpl<Uuid> threadProducerInstanceIdFuture = new KafkaFutureImpl<>();
+    private volatile KafkaFutureImpl<Uuid> producerInstanceIdFuture = new KafkaFutureImpl<>();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:352]: We only have producer per thread and can unify both cases (up to know, only one of both variables would be used, never both at the same time)

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 4]
```diff
@@ -806,46 +805,34 @@ void maybeGetClientInstanceIds() {
...
                     }
                 } else {
-                    threadProducerInstanceIdFuture.completeExceptionally(
+                    producerInstanceIdFuture.completeExceptionally(
                         new TimeoutException("Could not retrieve thread producer client instance id.")
                     );
                 }
             }
 
-            maybeResetFetchDeadline();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:833]: We only use this once, and inline this method with this PR (as the method itself is simplified)

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 5]
```diff
@@ -1610,7 +1597,7 @@ public Object getStateLock() {
     }
 
     // this method is NOT thread-safe (we rely on the callee to be `synchronized`)
-    public Map<String, KafkaFuture<Uuid>> consumerClientInstanceIds(final Duration timeout) {
+    public Map<String, KafkaFuture<Uuid>> clientInstanceIds(final Duration timeout) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:1600]: We unify consumer and producer in this single method now

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 6]
```diff
@@ -1685,27 +1680,6 @@ public void shouldThrowTimeoutExceptionWhenGlobalConsumerFutureDoesNotComplete()
         }
     }
 
-    @Test
-    public void shouldThrowTimeoutExceptionWhenThreadProducerFutureDoesNotComplete() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:1689]: No need to test producer isolates, as consumer/producer methods got unified.

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 7]
```diff
@@ -1813,14 +1738,9 @@ public Uuid get(final long timeout, final TimeUnit timeUnit) {
...
         assertThat(didAssertThreadOne.get(), equalTo(true));
         assertThat(didAssertThreadTwo.get(), equalTo(true));
-        assertThat(didAssertThreadProducer.get(), equalTo(true));
-        assertThat(didAssertTaskProducers.get(), equalTo(true));
-        assertThat(didAssertTask1.get(), equalTo(true));
-        assertThat(didAssertTask2.get(), equalTo(true));
         assertThat(didAssertGlobalThread.get(), equalTo(true));
     }
 
-    @Deprecated // testing old PAPI
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:1823]: Side cleanup -- this is not testing any old API -- removing unnecessary and incorrect annotation

**Review by bbejeck — APPROVED** [review]

**Pull request merged**